src/mds/mdstypes.h

```diff
@@ -627,6 +631,8 @@ struct inode_t {
   std::vector<uint8_t> fscrypt_auth;
   std::vector<uint8_t> fscrypt_file;
 
+  bufferlist fscrypt_last_block;
+
```

Comment:
Hmm, so this lives in the inode_t? I forget -- is inode_t a long-lived structure in the MDS that describes an inode or just a transient description of the inode that lives in a per-call fashion? If it's the former, then how is this value protected against racing truncate calls from different clients?

Reply:
It seems is, I will fix this and move this to a proper place.

Reply:
Currently the last block contents will be journaled together with the project inode and it will be cleared after the truncate finishes, and won't make it persistent together with the `CInode:inode_t` in the disks.